src/filesearch/MarkdownFileLocator.js

```diff
@@ -0,0 +1,39 @@
+const fs = require('fs');
+const path = require('path');
+
+const DESIRED_EXT = '.md';
+
+class MarkdownFileLocator {
+    locateFiles(basePath) {
+        if (false === fs.existsSync(basePath)) {
+            return [];
+        }
+
+        if(DESIRED_EXT === path.extname(basePath)) {
+            return [basePath];
+        }
+
+        return this._scanPath(basePath);
+    }
+
+    _scanPath(basePath) {
+        let ext = path.extname(basePath);
+        if('' !== ext && DESIRED_EXT !== ext) {
```

Comment:
Wouldn't it be better to check if the ext is not a different extension than empty or .md somewhere else and give out a clean error?

Kinda odd to give an empty array if I fave myfile.mdd (or if someone uses odd extensions for their markdowns although they still have fine syntax)

Reply:
https://superuser.com/questions/249436/file-extension-for-markdown-files

We should allow the ones present in this list, it would be less confusing for people who use them.

Reply:
Updated to match some common extensions.